packages/kit/src/lib/HookSystem.ts

```diff
@@ -176,24 +241,40 @@ export class HookSystem<
 		return hooks;
 	}
 
+	/**
+	 * Returns list of hooks for a given name
+	 */
+	hooksForName(name: string): RegisteredHook[] {
+		return this._registeredHooks[name] ?? [];
+	}
+
 	createScope<TExtraArgs extends unknown[] = never[]>(
 		owner: string,
 		extraArgs: [...TExtraArgs],
-		meta: Partial<RegisteredHookMeta> = {},
-	): CreateScopeReturnType<THookFns, TExtraArgs> {
+		scopeMeta: Partial<RegisteredHookMeta> = {},
+	): CreateScopeReturnType<THooks, TExtraArgs, THookFns, THookMetas> {
 		return {
 			hook: <TName extends Extract<keyof THookFns, string>>(
 				name: TName,
 				hookFn: WithExtraArgs<THookFns[TName], TExtraArgs>,
+				...args: HookMetaArg<THookMetas[TName]>
 			): void => {
 				const internalHook = ((...args: Parameters<THookFns[TName]>) => {
 					return hookFn(...args, ...extraArgs);
 				}) as THookFns[TName];
 
-				return this.hook(owner, name, internalHook, {
-					...meta,
+				const meta = {
+					...scopeMeta,
+					...args[0],
 					external: hookFn,
-				});
+				};
+
+				return this.hook(
+					owner,
+					name,
+					internalHook,
+					...([meta] as HookMetaArg<THookMetas[TName]>),
```

Comment:
For my own knowledge, what's going on here? Is the extra `...([])` necessary so you can assert the type?